src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -440,15 +439,202 @@ class BraketLocalAquilaDevice(BraketAhsDevice):
         device = LocalSimulator("braket_ahs")
         super().__init__(wires=wires, device=device, shots=shots)
 
+    def create_ahs_program(self, evolution):
+        """Create AHS program for simulation from a ParametrizedEvolution
+
+        Args:
+            evolution (ParametrizedEvolution): the PennyLane operator describing the pulse
+                to be converted into an Analogue Hamiltonian Simulation program
+
+        Returns:
+            AnalogHamiltonianSimulation: a program containing the register and drive
+                information for running an AHS task on simulation or hardware"""
+
+        # sets self.pulses to be the evaluated pulses (now only a function of time)
+        self._evaluate_pulses(evolution)
+        self._create_register(evolution.H.settings.register)
+
+        time_interval = evolution.t
+
+        H = self._convert_pulse_to_driving_field(self.pulses[self._global_pulse], time_interval)
+
+        # Create local detunings
+        local_detunings = self._create_valid_local_detunings()
+        if local_detunings is not None:
+            detuning, pattern = self._extract_pattern_from_detunings(local_detunings, time_interval)
+            shift = self._convert_pulses_to_shifting_field(detuning, pattern, time_interval)
+            H = H + shift
+
+        ahs_program = AnalogHamiltonianSimulation(register=self.register, hamiltonian=H)
+        self.ahs_program = ahs_program
+
+        return ahs_program
+
     def _run_task(self, ahs_program):
         task = self._device.run(ahs_program, shots=self.shots, steps=100)
         return task
+
+    def _create_valid_local_detunings(self):
+        """Return ordered list of local detunings for all wires in device.
+
+        Returns:
+            List[Union[callable, float]]: List of detunings covering all device wires.
+        """
+        local_pulses = self.pulses.copy()
+        local_pulses.pop(self._global_pulse)
+        if len(local_pulses) == 0:
+            return None
+
+        callable_detunings = callable(local_pulses[0].frequency)
+        device_detunings = (
+            [lambda t: 0] * len(self.wires) if callable_detunings else [0] * len(self.wires)
+        )
+
+        for pulse in local_pulses:
+            for wire in pulse.wires:
+                device_detunings[self.wires.index(wire)] = pulse.frequency
+
+        return device_detunings
+
+    def _extract_pattern_from_detunings(self, detunings, time_interval):
+        """Use the detunings as defined in PennyLane to find the pattern for the ``ShiftingField``
+
+        Args:
+            detunings (List[Union[float, callable]]): detunings to extract pattern from
+            time_interval(array[Number, Number]]): The start and end time for the applied pulses
+
+        Returns:
+            Union[float, callable]: Updated detuning for ``ShiftingField``
+            Pattern: Pattern object containing magnitude of detunings for individual atoms in the device
+        """
```

Comment:
A bit more explanation would be very helpful here since this was very confusing for us. Explain in more detail what this method is doing and why